src/crypto/DeviceList.js

```diff
@@ -210,134 +195,124 @@ export default class DeviceList {
      * @param {String} userId
      */
     invalidateUserDeviceList(userId) {
+        if (typeof userId !== 'string') {
+            throw new Error('userId must be a string; was '+userId);
+        }
         this._pendingUsersWithNewDevices[userId] = true;
     }
 
     /**
-     * Start device queries for any users who sent us an m.new_device recently
+     * Start device queries for any users with outdated device lists
+     *
+     * We tolerate multiple concurrent device queries, but only one query per
+     * user.
+     *
+     * If any users already have downloads in progress, they are ignored -
+     * they will be refreshed when the current download completes anyway.
```

Comment:
Hrm. I've expanded the comment abit, but perhaps I'm missing the source of confusion here?